### PR TITLE
test(signal_processing): fix unit test bug

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

Related to https://github.com/autowarefoundation/autoware.universe/issues/1959

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/signal_processing/include/signal_processing/lowpass_filter.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ class LowpassFilterInterface
 public:
   explicit LowpassFilterInterface(const double gain) : gain_(gain) {}
 
-  void reset() { x_ = T{}; }
+  void reset() { x_ = {}; }
   void reset(const T & x) { x_ = x; }
```
Comment (takayuki5168) [inline common/signal_processing/include/signal_processing/lowpass_filter.hpp:37]: Changed the same way as
https://github.com/autowarefoundation/autoware.universe/blob/4c739d64cc0de0264c1122c26c97dd3646758c3e/common/signal_processing/src/lowpass_filter_1d.cpp#L19

**Comment by kenji-miyake** [comment]
Thank you for your fixing.
Could I test CI before merging this PR?

**Comment by takayuki5168** [comment]
@kenji-miyake Of course. I'm not in a hurry.

**Comment by ambroise-arm** [comment]
LGTM, tested locally on arm

**Comment by kenji-miyake** [comment]
@ambroise-arm Could you review https://github.com/autowarefoundation/autoware-github-actions/pull/185 before merging this PR, please? :pray:

**Comment by takayuki5168** [comment]
@kenji-miyake I tested locally with RelWithDebInfo, the issue did not occur.

**Review by kenji-miyake — APPROVED** [review]
Since I was able to reproduce the issue in https://github.com/autowarefoundation/autoware.universe/pull/1965, it's okay to merge this PR.

**Comment by kenji-miyake** [comment]
@takayuki5168 Since this PR won't fix any feature, I've changed the title from `fix` to `test`.

**Comment by takayuki5168** [comment]
@kenji-miyake 
In my understanding, this PR resolved the issue that unit test failed on Arm. But should we set `test` instead of `fix`?
https://github.com/autowarefoundation/autoware.universe/issues/1959

**Comment by kenji-miyake** [comment]
@takayuki5168 Uh, sorry. I overlooked the first file. :pleading_face:

**Pull request merged**